skimage/feature/fisher_vector.py

```diff
@@ -0,0 +1,208 @@
+"""
+fisher_vector.py - Implementation of the Fisher vector encoding algorithm
+
+This module contains the source code for Fisher vector computation. The
+computation is separated into two distinct steps, which are called separately
+by the user, namely:
+
+learn_gmm: Used to estimate the GMM for all vectors/descriptors computed for
+           all examples in the dataset (e.g. estimated using all the SIFT
+           vectors computed for all images in the dataset, or at least a subset
+           of this).
+
+compute_fisher_vector: Used to compute the Fisher vector representation for a
+                       single set of descriptors/vector (e.g. the SIFT
+                       descriptors for a single image in your dataset, or
+                       perhaps a test image).
+
+Reference: Perronnin, F. and Dance, C. Fisher kernels on Visual Vocabularies
+           for Image Categorization, IEEE Conference on Computer Vision and
+           Pattern Recognition, 2007
+
+Origin Author: Dan Oneata (Author of the original implementation for the Fisher
+vector computation using scikit-learn and NumPy. Subsequently ported to
+scikit-image (here) by other authors.)
+"""
+import typing
+
+import numpy as np
+from sklearn.mixture import GaussianMixture
```

Comment:
I think this `GaussianMixture` import would need to be inside of `learn_gmm` so that we can have scikit-learn as an optional rather than required dependency.

Reply:
prior discussion: https://github.com/scikit-image/scikit-image/issues/4057#issuecomment-515235811

Reply:
Will do.

Also, CI is failing with the following error:
`ValueError: expected parenthesized list: '-learn( (>= 0.18.1))'`

I'm not sure how to fix that as I think setup.py in this case is expecting module names (i.e. the dash in scikit-learn is not being accepted)? @grlee77

Reply:
I added a comment about this there. If you make that change of moving it to requirements/optional.txt, you will have to avoid this top-level import here for CI to pass since only some CI runs install the optional dependencies.

Reply:
@grlee77 Done

Reply:
Done